src/measure_theory/ae_eq_fun.lean

```diff
@@ -2,13 +2,74 @@
 Copyright (c) 2019 Johannes Hölzl, Zhouhang Zhou. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johannes Hölzl, Zhouhang Zhou
-
-We define almost everywhere equal functions, and show that
-  • they form a vector space if the codomain is a vector space
-  • they form an emetric space under L¹ metric if the codomain is a metric space
 -/
+
 import measure_theory.integration
 
+/-!
+
+# ALmost everywhere equal functions
+
+Two measurable functions are treated as identical if they are almost everywhere equal. We form the
+set of equivalence classes under the relation of being almost everywhere equal, which is sometimes
+known as the L0 space.
+
+See `l1_space.lean` for L1 space.
+
+
+## Notation
+
+* `α →ₘ β` is the type of L0 space. `f : α →ₘ β` is a "function" in L0.
+  In comments, `[f]` is also used to denote an L0 function.
+
+  `ₘ` can be typed as `\_m`. Sometimes it is shown as a box if font is missing.
+
+
+## Main statements
+
+* The linear structure of L0 :
+    Addition and scalar multiplication are defined on L0 in the natural way, i.e.,
+    `[f] + [g] := [f + g]`, `c • [f] := [c • f]`. So defined, `α →ₘ β` inherits the linear structure
+    of `β`. For example, if `β` is a module, then `α →ₘ β` is a module over the same ring.
+
+    See `mk_add_mk`,  `neg_mk`,     `mk_sub_mk`,  `smul_mk`,
+        `add_to_fun`, `neg_to_fun`, `sub_to_fun`, `smul_to_fun`
+
+* The order structure of L0 :
+    `≤` can be defined in a similar way: `[f] ≤ [g]` if `f a ≤ g a` for almost all `a` in domain.
+    And `α →ₘ β` inherits the preorder and partial order of `β`.
+
+    TODO: Define `sup` and `inf` on L0 so that it forms a lattice. It seems that `β` must be a
+    linear order, since otherwise `f ⊔ g` may not be a measurable function.
+
+* Emetric on L0 :
+    If `β` is an `emetric_space`, then L0 can be made into an `emetric_space`, where
+    `edist [f] [g]` is defined to be `∫⁻ a, edist (f a) (g a)`.
+
+    The integral used here is `lintegral : (α → ennreal) → ennreal`, which is defined in the file
+    `integration.lean`.
+
+    See `edist_mk_mk` and `edist_to_fun`.
+
+
+## Implementation notes
+
+`f.to_fun`  : To find a representative of `f : α →ₘ β`, use `f.to_fun`.
+              For each operation `op` in L0, there is a lemma called `op_to_fun`, characterizing,
+              say, `(f op g).to_fun`.
+
+`mk`        : To constructs an L0 function from an ordinary function, use `mk`
```

Comment:
Rather `ae_eq_fun.mk`. Also, instead of an ordinary function, say `an ordinary measurable function`.